src/format.rs

```diff
@@ -44,6 +45,16 @@ impl Format {
 
         Some(Format(results))
     }
+
+}
+
+pub fn display_with<'a>(format: &'a Format,
```

Comment:
I turned it into a method of a private trait. I'm not entirely sure that the trait is appropriately generic, but given that it currently has a single use, I don't think that is a big deal.